src/mlpack/methods/quic_svd/quic_svd.hpp

```diff
@@ -67,10 +70,10 @@ class QUIC_SVD
    * @param epsilon Error tolerance fraction for calculated subspace.
    * @param delta Cumulative probability for Monte Carlo error lower bound.
    */
-  QUIC_SVD(const arma::mat& dataset,
+  void Apply(const arma::mat& dataset,
            arma::mat& u,
-           arma::mat& v,
            arma::mat& sigma,
+           arma::mat& v,
            const double epsilon = 0.03,
            const double delta = 0.1);
```

Comment:
I agree, I think this is a better way to structure this class in general.  But if we have `void Apply()`, I think we can actually remove the `dataset` member from the class itself.  To match the other classes a little better, I might suggest having the constructor take `epsilon` and `delta` as parameters instead of `Apply()`.

Also, I think it would be useful to templatize `Apply()` to accept any `MatType`, so it can work with both dense and sparse matrices.

Reply:
Even I didn't particularly think having dataset as a member is necessary, I just went with the existing changes. I'll make the changes :) .